src/main/perl/Lock.pm

```diff
@@ -202,15 +206,17 @@ sub set_lock {
   }
   # now got the lock (or forcing!)
   my $lf=FileHandle->new("> " . $self->{'LOCK_FILE'});
+  $lf->autoflush;
   unless ($lf) {
     $self->error("cannot create lock file: ".$self->{'LOCK_FILE'});
     return undef;
   }
-  print $lf $$;
-  unless ($lf->close()) {
-    $self->error("cannot close lock file: ".$self->{'LOCK_FILE'});
+  unless (flock ($lf, LOCK_EX|LOCK_NB)) {
```

Comment:
This is ignoring `FORCE_ALWAYS` and `FORCE_IF_STALE`, which means those options will be broken by this commit at present.